spock-core/src/main/java/org/spockframework/runtime/SpockRuntime.java

```diff
@@ -117,6 +118,15 @@ public static void verifyMethodCondition(@Nullable ErrorCollector errorCollector
     }
   }
 
+  private static boolean isVoidMethod(@Nullable Object target, String method, Object... args) {
+    if (target instanceof Closure) { // since we support verifyAll we must check the closure hierarchy
+      Closure closure = ((Closure)target);
+      return GroovyRuntimeUtil.isVoidMethod(closure.getDelegate(), method, args)
```

Comment:
Couldn't `delegate` be a `Closure` for nested `Closure`?

Reply:
(in general in Groovy if set programatically - I don't know it there is a real life scenario for that, but maybe calling `isVoidMethod()` instead of `GroovyRuntimeUtil.isVoidMethod()` is safer here)

Reply:
As I said I don't know if there is a real life scenario for that. It would need to be a nested Closure with a delegate programatically modified to be the outer `Closure`. I just put your attention that it is possible in Groovy to have a delegate which is `Closure`. I wonder if you see a potential problem with calling `this.isVoidMethod(closure.getDelegate(), ...)` instead (for in case)?

Reply:
Although it would be possible, I don't think that it would be "correct" from the syntax standpoint. This method is only called within `verifyAll` via `verifyMethodCondition` so I don't see a scenario where this comes to pass, nested closures are not decorated via `verifyMethodCondition` and if it is a method accepting the closure then the return type of the method is what we are looking at.

Reply:
Thanks for the explanation! Makes sense to me.

Reply:
The same for me.